### DISCUSSION TITLE
API Based Login - Not using Service Accounts

### DISCUSSION THREAD

**Discussion by thomasjbhayes** [goauthentik/authentik#7699] [Q&A]
Hello,

I am considering using Authentik to implement user management for an application. The application is a project I inherited that is already largely built, and has its own login page that I wish to maintain.

I am trying to implement a flow where I can use Authentik as an authentication server but maintaining the existing login UI. Specifically, I want to implement an API call where I can call Authentik from the app with user credentials (username & password) and receive my access token as a response.

So far I have found the machine-to-machine flow (https://goauthentik.io/docs/providers/oauth2/client_credentials) which works fine, but it is dependent on using a service account.

I do not wish to use service accounts with my application as I want users to be able to create their accounts and modify their passwords through the Authentik UI flows, however login to be managed using the existing app login flow.

Please let me know if there is any solution that can implement my desired approach with Authentik.

**Comment by arsiba** [comment]
Are there any follow-ups on this topic? I'm currently facing similar issues. Has anybody found a way to implement this?

**Comment by rtorrespr** [comment]
I just found an answer to a similar question [7228](https://github.com/goauthentik/authentik/discussions/7228).
